Makefile

```diff
@@ -21,27 +21,41 @@ UNITDIR              = unit
 UNIT_TESTSDIR        = $(TESTS_DIR)/$(UNITDIR)
 EXAMPLES_DIR         = examples
 
-SRC := $(shell find $(SRCDIR) -name "*.c")
+# Define a recursive wildcard function to 'find' all files under a sub-dir
+# See https://stackoverflow.com/questions/2483182/recursive-wildcards-in-gnu-make/18258352#18258352
+define rwildcard =
+	$(foreach d,$(wildcard $(1:=/*)),$(call rwildcard,$d,$2) $(filter $(subst *,%,$2),$d))
+endef
+
+SRC := $(call rwildcard, $(SRCDIR), *.c)
 
 # Generate list of common test source files, only from tests/ dir. Hence '-maxdepth 1'.
 # These objects are shared between functional/ and unit/ test binaries.
-COMMON_TESTSRC := $(shell find $(TESTS_DIR) -maxdepth 1 -name "*.c")
-FUNCTIONAL_TESTSRC := $(shell find $(FUNCTIONAL_TESTSDIR) -name "*.c")
+COMMON_TESTSRC := $(wildcard $(TESTS_DIR)/*.c)
+FUNCTIONAL_TESTSRC := $(wildcard $(FUNCTIONAL_TESTSDIR)/*.c)
```

Comment:
rwildcard is the most equivalent to the old code.

Reply:
I don't quite agree. On L35, we are hard-wiring the list of functional test sources to _only_ those that are in `functional/ test sub-dir, which currently has no further sub-dirs. So the search output is closer to what `wildcard` will provide.

In any case, I agree that it's better to use `rwildcard` here, and in similar such cases, to allow for the possibility that we may create sub-dirs under `functional/`, `unit/` and `examples/` sub-dirs.

With that in mind, I have revised the change to use `rwildcard` more often, whenever it is possible to do so, w/o breaking existing functionality.